File: JULY-2025/27-07-2025/p2210.java
class Solution {
    public int countHillValley(int[] nums) {
        int res = 0;
        for (int i = 1;i<nums.length-1;i++) {
            if (nums[i] == nums[i-1]) {continue;}

            int j = i+1;
            while(i<nums.length-1 && j<nums.length-1 && nums[i] == nums[j]) {
                j++;
            }
            if (nums[i] < nums[i-1] && nums[i] < nums[j]) {
                res++;
            }
            if (nums[i] > nums[i-1] && nums[i] > nums[j]) {
                res++;
            }
            i = j-1;
        }
        return res;
    }
}

File: JULY-2025/27-07-2025/p2210.cs
public class Solution {
    public int CountHillValley(int[] nums) {
        int pr = 0,cnt=0;
        for(int i=0;i<nums.Length-1;i++)
        {
            if((pr>0 && nums[i]>nums[i+1])||(pr<0 && nums[i]<nums[i+1]))cnt++;
            if(nums[i]!=nums[i+1])
             {
                if(nums[i]>nums[i+1]) pr=-1;
                                 else pr=1;
             }
        }
        return cnt;
    }
}